### PR TITLE
User model unit tests

### PR DESCRIPTION
**Pull request by ujwal-jain**
Added user model unit tests. Checks for missing required fields in the User and also an invalid short length password in the user.

To test, pull the unittests branch and run `npm run test`

### CODE REVIEW

File: backend/models/user.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      expect(err).toBeDefined();
+      expect(err.errors[key].message).toBe(`Path \`${key}\` is required.`);
+    }
+  });
+  it("should invalidate user when given short password", () => {
+    const user = validUser();
+    user["password"] = "invp";
+    const err = user.validateSync();
+    expect(err).toBeDefined();
+
```
Comment (SiddharthKartha) [inline backend/models/user.test.js]: Delete extra line

**Comment by SiddharthKartha** [comment]
>

**Comment by SiddharthKartha** [comment]
Everything else looks good to me.

**Comment by JihoPark21** [comment]
Looks good to me

**Pull request merged**